components/gitpod-cli/cmd/rebuild.go

```
@@ -0,0 +1,234 @@
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/gitpod-io/gitpod/gitpod-cli/pkg/supervisor"
+	"github.com/gitpod-io/gitpod/gitpod-cli/pkg/utils"
+	"github.com/gitpod-io/gitpod/supervisor/api"
+	"github.com/spf13/cobra"
+)
+
+func TerminateExistingContainer() error {
+	cmd := exec.Command("docker", "ps", "-q", "-f", "label=gp-rebuild")
+	containerIds, err := cmd.Output()
+	if err != nil {
+		return err
+	}
+
+	for _, id := range strings.Split(string(containerIds), "\n") {
+		if len(id) == 0 {
+			continue
+		}
+
+		cmd = exec.Command("docker", "stop", id)
+		err := cmd.Run()
+		if err != nil {
+			return err
+		}
+
+		cmd = exec.Command("docker", "rm", "-f", id)
+		err = cmd.Run()
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+var buildCmd = &cobra.Command{
+	Use:    "rebuild",
+	Short:  "Re-builds the workspace image (useful to debug a workspace custom image)",
+	Hidden: false,
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := context.Background()
+
+		client, err := supervisor.New(ctx)
+		if err != nil {
+			utils.LogError(ctx, err, "Could not get workspace info required to build", client)
+			return
+		}
+		defer client.Close()
+
+		wsInfo, err := client.Info.WorkspaceInfo(ctx, &api.WorkspaceInfoRequest{})
+		if err != nil {
+			utils.LogError(ctx, err, "Could not fetch the workspace info", client)
+			return
+		}
+
+		tmpDir, err := os.MkdirTemp("", "gp-rebuild-*")
+		if err != nil {
+			log.Fatal("Could not create temporary directory")
+			return
+		}
+		defer os.RemoveAll(tmpDir)
+
+		event := utils.TrackEvent(ctx, client, &utils.TrackCommandUsageParams{
+			Command: cmd.Name(),
+		})
+		defer event.Send(ctx)
+
+		gitpodConfig, err := utils.ParseGitpodConfig(wsInfo.CheckoutLocation)
+		if err != nil {
```

err is swallowed, we should give a root cause to the user